prusti-interface/src/environment/mir_utils/place_add_field.rs

```diff
@@ -0,0 +1,19 @@
+use std::iter::once;
+
+use rustc_middle::mir;
+use rustc_middle::ty;
+
+pub trait PlaceAddField<'tcx> {
+    fn field(self, tcx: ty::TyCtxt<'tcx>, field: mir::Field, ty: ty::Ty<'tcx>) -> mir::Place<'tcx>;
```

Comment:
I think [`tcx.mk_place_field`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/struct.TyCtxt.html#method.mk_place_field) does the same.